src/MongoDB.Driver.Core/Core/Misc/HintForUpdateAndReplaceOperationsFeature.cs

```diff
@@ -0,0 +1,51 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+namespace MongoDB.Driver.Core.Misc
+{
+    /// <summary>
+    /// Represents the hint for write operations feature.
+    /// </summary>
+    /// <seealso cref="MongoDB.Driver.Core.Misc.Feature" />
+    public class HintForUpdateAndReplaceOperationsFeature : Feature
+    {
+        private readonly SemanticVersion _exceptionThresholdVersion;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HintForUpdateAndReplaceOperationsFeature"/> class.
+        /// </summary>
+        /// <param name="name">The name of the feature.</param>
+        /// <param name="firstSupportedVersion">The first server version that supports the feature.</param>
+        /// <param name="exceptionThresholdVersion">The version below which throwing exception is allowed.</param>
+        public HintForUpdateAndReplaceOperationsFeature(string name, SemanticVersion firstSupportedVersion, SemanticVersion exceptionThresholdVersion)
+            : base(name, firstSupportedVersion)
+        {
+            _exceptionThresholdVersion = exceptionThresholdVersion;
+        }
+
+        /// <summary>
+        /// Determines whether a feature is supported by a version of the server.
+        /// </summary>
+        /// <param name="serverVersion">The server version.</param>
+        /// <param name="isExceptionAllowed">Determines whether the driver can throw exception.</param>
+        /// <returns>Whether a feature is supported by a version of the server.</returns>
+        public bool IsSupported(SemanticVersion serverVersion, out bool isExceptionAllowed)
```

Comment:
This method is mixing two things. Let's use the inherited `IsSupported` method unchanged and add the following new method in this subclass:

    /// <summary>
    /// Determines whether the driver must throw an exception if the feature is not supported by the server.
    /// </summary>
    /// <param name="serverVersion">The server version.</param>
    /// <returns>Whether the driver must throw if feature is not supported.</returns>
    public bool DriverMustThrowIfNotSupported(SemanticVersion serverVersion)
    {
        return serverVersion < _v3_4_0;
    }

where `_v3_4_0`is a new field replacing `_exceptionThresholdVersion`:

    private readonly SemanticVersion _v3_4_0 = new SemanticVersion(3, 4, 0);

Reply:
Done.